bundles/org.polymodel.algebra/src/org/polymodel/algebra/prettyprinter/algebra/ISLPrettyPrinter.java

```diff
@@ -73,7 +76,25 @@ public void visitIntConstraintSystem(IntConstraintSystem i) {
 	}
 
 	public void visitAffineExpression(AffineExpression a) {
-		separateAffine(a.getTerms());
+		EList<AffineTerm> affineTerms = a.getTerms();
+		preprendExistential(affineTerms);
+		separateAffine(affineTerms);
+	}
+	
+	public void preprendExistential(EList<AffineTerm> terms) {
+		ArrayList<String> existentials = new ArrayList<>();
+		for (AffineTerm term : terms) {
+			Variable var =  term.getVariable();
+			// Not the cleanest way to check if variable is existential
+			// but we cannot import org.polymodel.ExistentialVariable
+			// without creating a circular build dependence
+			if (var != null && var.getClass().toString().contains("Existential")) { 
```

Comment:
Ideally, we would want to do something like check:
```var instanceof org.polymodel.ExistentialVariable```
but this would add a cyclic dependence between the org.polymodel and org.polymodel.algebra packages and will not build.